Dockerfile

```diff
@@ -1,33 +1,23 @@
-# Build the manager binary
-FROM golang:1.20 as builder
-ARG TARGETOS
-ARG TARGETARCH
+ARG BASE_IMAGE=scratch
+####################################################################################################
+# base
+####################################################################################################
+FROM alpine:3.17 as base
+ARG ARCH
+RUN apk update && apk upgrade && \
+    apk add ca-certificates && \
+    apk --no-cache add tzdata
 
-WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
+COPY dist/numaplane /bin/numaplane
 
-# Copy the go source
-COPY cmd/main.go cmd/main.go
-COPY api/ api/
-COPY internal/controller/ internal/controller/
+RUN chmod +x /bin/numaplane
 
-# Build
-# the GOARCH has not a default value to allow the binary be built according to the host where the command
-# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
-# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
-# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
-RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
-
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
-WORKDIR /
-COPY --from=builder /workspace/manager .
-USER 65532:65532
-
-ENTRYPOINT ["/manager"]
```

Comment:
Why did you end up changing the Dockerfile?

Reply:
If there's a reason to change it, maybe it should be in a separate PR? Also, please make sure to describe anything you do (in the PR main description and/or inline) if it helps to review.

Reply:
Okay, thanks! And nice job getting that working. Out of curiosity, what type of machine and OS do you have?

Reply:
macOS

Reply:
Do you have an M1 (arm64) or amd64?

Reply:
I just tried building the image myself and failed, so that would be great for you to add a separate PR. I have an M1 - not sure if you have the same. We can all try your PR and verify it's working for us. Thanks!